xbmc/activity/ActivityLog.h

```diff
@@ -0,0 +1,80 @@
+#pragma once
+/*
+ *      Copyright (C) 2014 Team XBMC
+ *      http://xbmc.org
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "activity/IActivity.h"
+#include "settings/lib/ISettingCallback.h"
+#include "threads/CriticalSection.h"
+
+#define NOTIFICATION_DISPLAY_TIME 5000
+#define NOTIFICATION_MESSAGE_TIME 1000
+
+typedef std::vector<ActivityPtr> Activities;
+
+class CActivityLog : public ISettingCallback
+{
+public:
+  virtual ~CActivityLog();
+
+  static CActivityLog& GetInstance();
+
+  Activities Get() const;
+  Activities Get(ActivityLevel level, bool includeHigherLevels = false) const;
+  ActivityPtr Get(const std::string& activityIdentifier) const;
+
+  void Add(ActivityPtr activity);
+  void Add(ActivityPtr activity, bool withNotification, bool withSound = true);
```

Comment:
It is but there are cases where depending on a condition (e.g. a setting) I either only want the activity or I want it with a notification so I added this "wrapper" method to not have to write a full if-else in multiple places.